src/lib/viewers/text/CSVViewer.js

```diff
@@ -53,38 +52,31 @@ class CSVViewer extends TextBaseViewer {
     load() {
         super.load();
 
-        const { representation, location } = this.options;
+        const { representation } = this.options;
         const template = representation.content.url_template;
-        const assetUrlCreator = createAssetUrlCreator(location);
-        const papaWorkerUrl = assetUrlCreator(JS[0]);
 
         return Promise.all([this.loadAssets(JS), this.getRepStatus().getPromise()])
             .then(() => {
-                this.api.get(papaWorkerUrl, { type: 'blob' }).then(papaWorkerBlob => {
-                    /* global Papa */
-                    const workerSrc = URL.createObjectURL(papaWorkerBlob);
-                    Papa.SCRIPT_PATH = workerSrc;
-
-                    this.startLoadTimer();
-                    const urlWithAuth = this.createContentUrlWithAuthParams(template);
-                    Papa.parse(urlWithAuth, {
-                        download: true,
-                        error: (err, file, inputElem, reason) => {
-                            const error = new PreviewError(ERROR_CODE.LOAD_CSV, __('error_refresh'), { reason });
-                            this.handleDownloadError(error, urlWithAuth);
-                        },
-                        complete: results => {
-                            if (this.isDestroyed() || !results) {
-                                return;
-                            }
-
-                            this.checkForParseErrors(results);
-
-                            this.data = results.data;
-                            this.finishLoading();
-                            URL.revokeObjectURL(workerSrc);
-                        },
-                    });
+                this.startLoadTimer();
+                const urlWithAuth = this.createContentUrlWithAuthParams(template);
+                Papa.parse(urlWithAuth, {
+                    delimitersToGuess: [',', '\t'],
```

Comment:
I've added the option `delimitersToGuess` which will restrict the delimiters to comma and tab which technically was all that we supported given that we support the extensions CSV and TSV. I could remove this option entirely, but then there is an issue with the auto-detection of delimiter for XSS.csv with the excessive `;` as well as `,`. 

At the end of the day it feels like keeping the delimiters to what we support makes sense. Allowing users to hack the system by naming it csv but providing delimiters such as `|` or `;` doesn't make all that much sense, imo.

Thoughts?

Reply:
Gotcha, so it sounds like we may need some new options to maintain consistency with the current experience. Should we update the value to `[',', '\t', Papa.RECORD_SEP, Papa.UNIT_SEP]` to retain support for the "true" delimiters mentioned in the docs?

Reply:
Changed version to 5.0.0 which doesn't contain the tweaks to the method which aims to guess the delimiter. Will follow up with Papaparse with the issue we're seeing